src/Arcus.EventGrid.Testing/Infrastructure/Hosts/EventConsumerHost.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using Polly;
+
+namespace Arcus.EventGrid.Testing.Infrastructure.Hosts
+{
+    public class EventConsumerHost
+    {
+        private static readonly Dictionary<string, string> _receivedEvents = new Dictionary<string, string>();
+        protected readonly ILogger _logger;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="logger">Logger to use for writing event information</param>
+        public EventConsumerHost(ILogger logger)
+        {
+            Guard.NotNull(logger, nameof(logger));
+
+            _logger = logger;
+        }
+
+        /// <summary>
+        ///     Handles new events that are being received
+        /// </summary>
+        /// <param name="rawReceivedEvents">Raw payload containing all events</param>
+        protected static void EventsReceived(string rawReceivedEvents)
+        {
+            var parsedEvents = JArray.Parse(rawReceivedEvents);
+            foreach (var parsedEvent in parsedEvents)
+            {
+                var eventId = parsedEvent["Id"]?.ToString();
```

Comment:
If we don't find `Id`, I would assume this is more of an error, and we shouldn't add a `null` key to `_receivedEvents` with the value of `rawReceivedEvents`?

Reply:
If it's not there we should throw an exception, I think, as we don't know what it is. Would that make sense?

Reply:
If it's not there we should throw an exception, I think, as we don't know what it is. Would that make sense?

Reply:
Feel free to let me know, I'll reflect this in a follow-up PR